php/Factory.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace PHPCD;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\Config\FileLocator;
+use Symfony\Component\DependencyInjection\Loader\YamlFileLoader;
+use Lvht\MsgpackRpc\JsonMessenger;
+use Lvht\MsgpackRpc\Io;
+use Lvht\MsgpackRpc\MsgpackMessenger;
+
+/**
+ * Simple factory to separate details of object creation
+ */
+class Factory
+{
+    /**
+     * @return ContainerBuilder
+     */
+    public function createDIContainer($configFileName, $configDir, $additionalParameters = [])
+    {
+        $container = new ContainerBuilder;
+
+        $loader = new YamlFileLoader($container, new FileLocator($configDir));
```

Comment:
Does `YamlFileLoader` is really better than `PHPFileLoader`? I don't think we should introduce the **Yaml Component** dependence.